cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -208,6 +208,38 @@ macro(ov_arm_neon_fp16_optimization_flags flags)
     endif()
 endmacro()
 
+#
+# ov_arm_sve_optimization_flags(<output flags>)
+#
+macro(ov_arm_sve_optimization_flags flags)
+    if(OV_COMPILER_IS_INTEL_LLVM)
+        message(WARNING "Unsupported CXX compiler ${CMAKE_CXX_COMPILER_ID}")
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        # nothing to define; works out of box
+    elseif(ANDROID)
+        if(ANDROID_ABI STREQUAL "arm64-v8a")
+            set(${flags} -march=armv8-a+sve -Wno-unused-command-line-argument)
+        else()
+            message(WARNING "SVE is not supported on this Android ABI: ${ANDROID_ABI}")
+        endif()
+    else()
+        if(AARCH64)
+            set(${flags} -O2 -march=armv8-a+sve)
+            if(NOT CMAKE_CL_64)
+                list(APPEND ${flags} -ftree-vectorize)
+            endif()
+            # Check for SVE support
+            if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+                set(${flags} ${${flags}} -march=armv8-a+sve)
```

Comment:
do we need some checks on compiler version? I suppose older compilers like gcc 9.x do not support SVE, do they?

Reply:
Would something like this work? Please tell me if there's a better way to check that.
```cmake
if(AARCH64)
    set(${flags} -O2 -march=armv8-a+sve)
    if(NOT CMAKE_CL_64)
        list(APPEND ${flags} -ftree-vectorize)
    endif()
    # Check for SVE support
    if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
        string(REGEX MATCH "^([0-9]+)\\.([0-9]+)" COMPILER_VERSION_MATCH ${CMAKE_CXX_COMPILER_VERSION})
        if (COMPILER_VERSION_MATCH)
            set(MAJOR_VERSION "${CMAKE_MATCH_1}")
            set(MINOR_VERSION "${CMAKE_MATCH_2}")
            
            # GCC 10.2 and higher support SVE
            if(${MAJOR_VERSION} GREATER_EQUAL 10 AND ${MINOR_VERSION} GREATER_EQUAL 2)
                set(${flags} ${${flags}} -march=armv8-a+sve)
            else()
                message(WARNING "SVE is not supported on compiler version ${CMAKE_CXX_COMPILER_VERSION}.")
            endif()
        else()
            message(WARNING "Could not determine compiler version.")
        endif()
    else()
        message(WARNING "SVE is not supported on this architecture.")
    endif()
elseif(ARM)
    message(WARNING "SVE is not supported on 32-bit ARM architectures.")
endif()
```